store/entity-store.html

```diff
@@ -84,92 +129,102 @@
+						headers: headers
+					})
+						.then(function(response) {
+							if (response.ok) {
+								return response.json();
+							}
+							return Promise.reject(response.status);
+						}.bind(this))
+						.then(function(body) {
+							var entity = window.D2L.Hypermedia.Siren.Parse(body);
+							this.update(entityId, tokenValue, entity);
+							if (bypassCache) {
+								this._invalidationListeners.forEach(function(listener) {
+									listener(entityId, cacheKey, entity);
+								});
+							}
+							return this._store.get(lowerCaseCacheKey).get(lowerCaseEntityId);
+						}.bind(this))
+						.catch(function(err) {
+							this.setError(entityId, tokenValue, err);
+							return this._store.get(lowerCaseCacheKey).get(lowerCaseEntityId);
+						}.bind(this));
+
+					this._store.get(lowerCaseCacheKey).set(lowerCaseEntityId, {
+						status: 'fetching',
+						entity: null,
+						request: request
+					});
 
-				return request;
-			}
+					return request;
+				}
 
-			if (entity.request) {
-				return entity.request;
-			} else {
-				return new Promise(function(resolve) {
-					this._notify(lowerCaseEntityId, lowerCaseToken, entity.entity);
-					resolve(entity);
-				}.bind(this));
-			}
+				if (entity.request) {
+					return entity.request;
+				} else {
+					return new Promise(function(resolve) {
+						this._notify(lowerCaseEntityId, cacheKey, entity.entity);
+						resolve(entity);
+					}.bind(this));
+				}
+			}.bind(this));
 		},
 
 		get: function(entityId, token) {
-			const lowerCaseToken = token.toLowerCase();
-			const lowerCaseEntityId = entityId.toLowerCase();
-
-			var entity = this._initContainer(this._store, entityId, token);
-			if (entity) {
-				return entity.entity;
-			} else {
-				return null;
-			}
+			return this._getToken(token).then(function (resolved) {
+				const cacheKey = resolved.cacheKey;
+
+				var entity = this._initContainer(this._store, entityId, cacheKey);
+				if (entity) {
+					return entity.entity;
+				} else {
+					return null;
+				}
+			}.bind(this));
 		},
 
 		update: function(entityId, token, entity) {
-			const lowerCaseToken = token.toLowerCase();
-			const lowerCaseEntityId = entityId.toLowerCase();
+			return this._getToken(token).then(function (resolved) {
+				const cacheKey = resolved.cacheKey;
 
-			this._initContainer(this._store, lowerCaseEntityId, lowerCaseToken);
-			return new Promise(function(resolve) {
-				const entities = this.expand(entityId, entity);
-				entities.forEach(function(entity) {
-					this._store.get(lowerCaseToken).set(entity.key.toLowerCase(), {
-						status: '',
-						entity: entity.value,
-						request: null
-					});
-					this._notify(entity.key, token, entity.value);
-				}.bind(this));
+				const lowerCaseCacheKey = cacheKey.toLowerCase();
+				const lowerCaseEntityId = entityId.toLowerCase();
+
+				this._initContainer(this._store, lowerCaseEntityId, lowerCaseCacheKey);
+				return new Promise(function(resolve) {
```

Comment:
nit: you can get rid of this `new Promise` and just `return entity;` since we're in a `Promise` now